### ISSUE TITLE
🏗️ [Core Feature]: Port v3 changes back into main

### ISSUE THREAD

**Issue by peektism** [OpenZeppelin/polkadot-runtime-templates#386]
### templates

- [X] Generic Runtime Template
- [X] Evm Template

### What is the key feature we're aiming to implement?

After #385 is merged and the release is made, we need to port the patches from `v3` branch back into main.

Maybe reopen https://github.com/OpenZeppelin/polkadot-runtime-templates/pull/381 and merge into main since latest changes. Might make more sense to open a fresh PR since https://github.com/OpenZeppelin/polkadot-runtime-templates/pull/385 will resolve most of the merge conflicts in https://github.com/OpenZeppelin/polkadot-runtime-templates/pull/381

### Contribution Guidelines

- [X] I agree to follow this project's Contribution Guidelines
